sys/net/link_layer/ieee802154/ieee802154_phy.c

```diff
@@ -0,0 +1,57 @@
+#include "net/netdev.h"
+#include "net/netdev/ieee802154.h"
+
+#include "net/netbuf.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* The transceiver HAL will call this function when there's a packet
+ * in the framebuffer of the transceiver. The role of the PHY layer
+ * is to use the transceiver HAL and the netbuf API to:
+ * - Allocate a buffer from the network stack
+ * - Read L1 data (LQI, RSSI, etc).
+ * - Generate a PHY indication with the packet + L1 data.
+ *
+ * Here we use some components of `netdev` as our transceiver
+ * HAL.
+ * Note this function is the same for all IEEE802154 devices,
+ * as seen, we reuse the PHY layer for those :)
+ */
+void ieee802154_rf_rx_done(netdev_t *dev)
+{
+    netdev_ieee802154_rx_info_t rx_info;
+    int bytes_expected = dev->driver->recv(dev, NULL, 0, NULL);
+    ieee802154_phy_ind_t ind = {0};
+
+    void *psdu = netbuf_alloc(bytes_expected, &ind.ctx);
+
+    if (ind.ctx == NULL) {
```

Comment:
`psdu` should be checked here. The stack might not provide context, right?

Reply:
I forgot to update this function after changing the prototype of `netbuf_alloc`. Indeed it's a check of `psdu`.